src/contracts/identity.ts

```diff
@@ -67,6 +72,25 @@ export const createBeaconProxyWithOwner = async (
   return contract.createBeaconProxyWithOwner(beacon, owner);
 };
 
+/**
+ * Create a new identity and register it to a handle to get a new DSNP Id.
+ * This will create and register a new beacon proxy identity.
+ *
+ * @param userAddress User's public key address
+ * @param handle The string handle to register
+ * @returns The contract Transaction
+ */
+export const createAndRegisterBeaconProxy = async (
+  userAddress: EthereumAddress,
+  handle: string
+): Promise<ContractTransaction> => {
+  const beaconFactory = await getBeaconFactoryContract();
+  const beaconAddr = await getBeaconAddress();
+  const { signer } = getConfig();
+  if (!signer) throw MissingSigner;
+  return await beaconFactory.connect(signer).createAndRegisterBeaconProxy(beaconAddr, userAddress, handle);
```

Comment:
Do we need the `.connect(signer)` here? I think `getBeaconFactoryContract` returns a contract instance that should already be connected to?.

Reply:
We do not! Nice catch.